habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/PyenvAndPoetrySetup.java

```diff
@@ -34,6 +34,10 @@ public class PyenvAndPoetrySetup {
      */
     static final String PYTHON_DEFAULT_VERSION_REQUIREMENT = "3.11.4";
 
+    public static boolean alreadyValidatedPoetryInstallation;
+    public static String alreadyValidatedPoetryVersion;
+    public static String priorActivePythonVersionActivated;
```

Comment:
Q: Is Habushu generally not marked thread-safe and/or do we not have have to worry about any race conditions for these checks?

Reply:
That's a good question. To be sure, I can add a synchronized block to the update for Poetry. The Python version can be added to a ThreadLocal to ensure consistency across the threads.  

We could also mark it explicitly as not thread-safe, as we rarely see anyone do a multi-threaded build due to the complexity of debugging it.

But it's easy enough to support this, so I'll make the updates.